File: backend/meetmind_ai_backend/app/Models/TaskCandidate.php
<?php

namespace App\Models;

use App\Enums\TaskCandidateStatus;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class TaskCandidate extends Model
{
    protected $fillable = [
        'meeting_id',
        'transcript_id',
        'title',
        'description',
        'suggested_assignee_name',
        'suggested_assignee_user_id',
        'suggested_deadline',
        'suggested_priority',
        'status',
    ];

    protected function casts(): array
    {
        return [
            'suggested_deadline' => 'date',
            'status' => TaskCandidateStatus::class,
        ];
    }

    public function meeting(): BelongsTo
    {
        return $this->belongsTo(Meeting::class);
    }

    public function transcript(): BelongsTo
    {
        return $this->belongsTo(Transcript::class);
    }

    public function suggestedAssignee(): BelongsTo
    {
        return $this->belongsTo(User::class, 'suggested_assignee_user_id');
    }

    public function confirmedTask(): BelongsTo
    {
        return $this->belongsTo(Task::class, 'confirmed_task_id');
    }
}

File: lib/features/tasks/domain/entities/task_candidate.dart
import 'package:equatable/equatable.dart';

/** FR-6.3. Mirrors `App\Enums\TaskCandidateStatus`. */
enum TaskCandidateStatus { pending, confirmed, dismissed }

/// An AI-extracted action item awaiting human confirmation before it
/// becomes a real Task — SRD FR-6.3: "Users shall review and confirm/edit
/// AI-extracted tasks before they are created."
class TaskCandidate extends Equatable {
  const TaskCandidate({
    required this.id,
    required this.meetingId,
    required this.title,
    required this.suggestedPriority,
    required this.status,
    this.description,
    this.suggestedAssigneeName,
    this.suggestedAssigneeUserId,
    this.suggestedDeadline,
  });

  final String id;
  final String meetingId;
  final String title;
  final String? description;

  /// Display name for the suggested assignee — the matched workspace
  /// member's name if the AI's free-text guess matched one, else the raw
  /// name it extracted from the transcript. Either way, just a suggestion
  /// for the reviewer; confirming can override it.
  final String? suggestedAssigneeName;
  final String? suggestedAssigneeUserId;
  final DateTime? suggestedDeadline;
  final String suggestedPriority;
  final TaskCandidateStatus status;

  @override
  List<Object?> get props => [
        id,
        meetingId,
        title,
        description,
        suggestedAssigneeName,
        suggestedAssigneeUserId,
        suggestedDeadline,
        suggestedPriority,
        status,
      ];
}
